hfile_s3.c

```diff
@@ -467,29 +467,25 @@ static char *escape_query(const char *qs) {
              c == '_' || c == '-' || c == '~' || c == '.' || c == '/' || c == '=' || c == '&') {
             escaped[j++] = c;
         } else {
-            sprintf(escaped + j, "%%%02X", c);
+            snprintf(escaped + j, alloced - j, "%%%02X", c);
             j += 3;
         }
     }
 
-    if (i != length) {
-        // in the case of a '?' copy the rest of the qs across unchanged
-        strcpy(escaped + j, qs + i);
-    } else {
-        escaped[j] = '\0';
-    }
```

Comment:
I noticed this while looking at it a few days ago.  It's an incorrect copy from escape_path, irrelevant here because it doesn't have the `if (c == '?') break` clause.

However it should probably get a mention in the commit message.

Reply:
Now mentioned.